docs-src/typescript/chapter-durable-execution/intrinsic-non-determinism.md

```diff
@@ -0,0 +1,34 @@
+---
+id: non-deterministic-branching-logic
+title: Non-deterministic branching logic
+description: History Replay, sometimes also called Workflow Replay, is the mechanism that Temporal uses to reconstruct the state of a Workflow Execution. Temporal provides Durable Execution via this Replay Functionality.
+sidebar_label: Durability through Replays
+tags:
+  - go sdk
+  - developer-guide-doc-type
+  - event history
+  - replay
+  - durable execution
+---
+
+Also referred to as "intrinsic non-determinism", writing bad branching logic in Workflow code prevents the Workflow code from executing to completion because the Workflow can have a different code path than the one expected from the Event History.
+
+### Common sources of non-determinism
+
+- Using Random numbers
```

Comment:
Only the Run ID thing is true in TS SDK. It would be useful to change the phrasing a bit to make it clear these are not actual concerns in TS, though it is useful to explain what could be sources of non-determinism if we didn't have the sandbox (so they understand why we do have the sandbox, and why some packages can't be used inside the sandbox)...